ci/skeleton/usr/share/hhvm/uninstall_fastcgi.sh

```diff
@@ -0,0 +1,219 @@
+#!/bin/bash
+
+apache_check_installed() {
+	echo "Checking if Apache is installed"
+	if [ \( ! -d /etc/apache2/mods-enabled \) -o \( ! -d /etc/apache2/mods-available \) ]
+	then
+		echo "WARNING: Couldn't find Apache2 configuration paths, not configuring"
+		return 1
+	fi
+	echo "Detected Apache installation"
+	return 0
+}
+
+apache_check_custom() {
+	echo "Looking for custom proxy configuration"
+	if [ ! -z "$(grep 9000 -R /etc/apache2/mods-enabled | grep ProxyPass)" ]
+	then
+		echo "WARNING: Clashing existing configuration detected, not configuring"
+		return 0
+	fi
+	echo "No custom proxy configuration found"
+	return 1
+}
+
+apache_check_module() {
+	echo "Checking for ${2} ${1}.${3}"
+	if [ ! -f "/etc/apache2/mods-${2}/${1}.${3}" ]
+	then
+		echo "Not found"
+		return 1
+	fi
+	echo "Found, checking for loading directives"
+	if [ "$3" == "load" ]
+	then
+		result=$(grep -E "^[^#]*LoadModule\\s*${1}_module" "/etc/apache2/mods-${2}/${1}.${3}")
+ 		if [ -z "$result" ]
+		then
+			"Loading directive not found"
+			return 1
+		fi
+	fi
+	echo "Detected ${2} ${1}.${3} configuration, setting up integration"
+	return 0
+}
+
+apache_force_enable_module() {
+	echo "Force enabling module ${1}.${2}"
+	if [ ! -f "/etc/apache2/mods-available/${1}.${2}" ]
+	then
+		echo "WARNING: Unsupported ${1}, not configuring"
+		return 1
+	fi
+	echo "Available module found"
+	rm -f "/etc/apache2/mods-enabled/${1}.${2}"
+	echo "Removed possible duplicates"
+	apache_enable_module $1 $2
+	echo "Completed force enabling"
+	return 0
+}
+
+apache_enable_module() {
+	echo "Enabling module ${1}.${2}"
+	if [ ! -f "/etc/apache2/mods-available/${1}.${2}" ]
+	then
+		echo "WARNING: Could not enable ${1}, not configuring"
+		return 1
+	fi
+	echo "Found available module"
+	if [ -f "/etc/apache2/mods-enabled/${1}.${2}" ]
+	then
+		echo "Module already enabled"
+		return 0
+	fi
+	echo "Creating a symlink"
+	ln -s "../mods-available/${1}.${2}" "/etc/apache2/mods-enabled/${1}.${2}"
+	echo "Finished creating a symlink"
+	return 0
+}
+
+apache_disable_module() {
+	echo "Disabling a module ${1}.${2}"
+	rm -f "/etc/apache2/mods-enabled/${1}.${2}"
+	echo "Finished disabling a module"
+}
+
+apache_restart() {
+	echo "Restarting apache"
+	if [ ! -f "/etc/init.d/apache2" ]
+	then
+		echo "Apache init.d script not detected"
+		return 0
+	fi
+	result=$(/etc/init.d/apache2 status | grep 'is running')
+	if [ ! -z "$result" ]
+	then
+		echo "Apache is running, restarting"
+		/etc/init.d/apache2 restart
+		echo "Finished restarting"
+	fi
+	echo "Finished restarting apache"
+	return 0
+}
+
+#!/bin/bash
```

Comment:
I know you copied this from the packaging repo, but it's weird. Is this a packaging convention or a typo @fredemmott?

Reply:
Definitely not a convention; these are just (possibly before my time  😲 ) convenience scripts

I suspect this was just someone combining into one years back and not noticing this.